libcxx/include/__type_traits/operation_traits.h

```diff
@@ -21,6 +21,7 @@ _LIBCPP_BEGIN_NAMESPACE_STD
 // Tags to represent the canonical operations
 struct __equal_tag {};
 struct __plus_tag {};
+struct __totally_ordered_less_tag {};
```

Comment:
I think I would rather use `__less_tag` here and continue having a 1-1 correspondence between these tags and normal C++ operators. I would potentially use a different mechanism to attach "additional semantics" to operations, once you actually have a need for that.